src/simplewallet/simplewallet.cpp

```diff
@@ -3368,6 +3368,10 @@ bool simple_wallet::init(const boost::program_options::variables_map& vm)
           } while (might_be_partial_seed(m_electrum_seed));
         }
       }
+      
+      std::string electrum_data = std::string(m_electrum_seed.data());
```

Comment:
This should be a reference so you don't malloc the copy.

Reply:
This does not copy, only calls a single constructor, correct?

Reply:
std::string on a c string causes a copy of the contents into the buffer? Wasn't referring to this "invokes the copy constructor" if that's what you mean?

Reply:
I guess I misunderstood you comment. I'm still not quite sure what you are suggesting though: if we don't create `std::string`, we can't use its handy methods below. Seems like a premature optimisation to me. But I'm fine with whatever you decide here, so just ignore my comment.

Reply:
The line below would just be, instead of using the extra copy here.

```std::string electrum_seed = m_electrum_seed.substr(0, m_electrum_seed.find_first_of("1234567890")-1);```

Reply:
But `m_electrum_seed` is a `epee::wipeable_string` which doesn't have `substr` or `find_first_of`, am I missing something?

Reply:
Not that it really matters since we're using a word now, but the type conversion is for the reason msgmaxim just posted

Reply:
Well that explains the confusion